src/relay.rs

```diff
@@ -293,6 +376,144 @@ impl<T: DuplexTransport + 'static> Network<T> {
             })
     }
 
+    /// Returns a Future to retrieve the withdrawal event for this transaction
+    ///
+    /// # Arguments
+    ///
+    /// * `transfer` - A Transfer struct with the tx_hash, block_hash, and block_number we need to retrieve the withdrawal
+    pub fn get_withdrawal_future(&self, transfer: &Transfer) -> impl Future<Item = Withdrawal, Error = ()> {
+        let tx_hash = transfer.tx_hash.clone();
+        let block_hash = transfer.block_hash.clone();
+        let block_number: &mut[u8] = &mut vec![0; 32];
+        transfer.block_number.to_big_endian(block_number);
+        let mut grouped: Vec<u8> = Vec::new();
+        grouped.extend_from_slice(&tx_hash[..]);
+        grouped.extend_from_slice(&block_hash[..]);
+        grouped.extend_from_slice(block_number);
+        let hash = H256(keccak256(&grouped[..]));
+        let account = self.account.clone();
+        self.relay.query::<Withdrawal, Address, BlockNumber, H256>(
+                "withdrawals",
+                hash,
+                account,
+                Options::default(),
+                BlockNumber::Latest,
+        ).or_else(|e| {
+            error!("error getting withdrawal: {}", e);
+            Err(())
+        })
+    }
+
+    /// Returns a Stream of Transfer that were missed, either from downtime, or failed approvals
+    ///
+    /// # Arguments
+    ///
+    /// * `handle` - Handle to the event loop to spawn additional futures
+    pub fn missed_transfer_stream(&self, interval: reactor::Interval, handle: &reactor::Handle) -> impl Stream<Item = Transfer, Error = ()> {
+        let (tx, rx) = mpsc::unbounded();
+        let handle = handle.clone();
+        let h = handle.clone();
+        let token_address = self.token.address().clone();
+        let relay_address = self.relay.address().clone();
+        let network_type = self.network_type;
+        let web3 = self.web3.clone();
+        let future = interval.for_each(move |_| {
+            let handle = h.clone();
+            let tx = tx.clone();
+            let web3 = web3.clone();
+            let w3 = web3.clone();
```

Comment:
I get why `h` and `w3` exist but i think there's probably a cleaner way to work around the borrow checker here, gonna run out of variable names at that rate.